c10/cuda/CUDAStream.cpp

```diff
@@ -304,7 +306,47 @@ LeakyStreamInternals* CUDAStream_internals(CUDAStream s) {
   }
 }
 
+void check_assert_state(const LeakyStreamInternals* ptr) {
```

Comment:
CPU multithreaded execution of CUDA kernels is a thing. What's the multithreading discipline here?